source/slang/slang.cpp

```diff
@@ -546,11 +546,24 @@ SlangResult Session::_readBuiltinModule(ISlangFileSystem* fileSystem, Scope* sco
     return SLANG_OK;
 }
 
-ISlangUnknown* Session::getInterface(const Guid& guid)
+SLANG_NO_THROW SlangResult SLANG_MCALL Session::queryInterface(SlangUUID const& uuid, void** outObject)
 {
-    if(guid == ISlangUnknown::getTypeGuid() || guid == IGlobalSession::getTypeGuid())
-        return asExternal(this);
-    return nullptr;
+    if (uuid == ISlangInternal::getTypeGuid())
+    {
+        // Special case to cast directly into internal type
```

Comment:
Why not addref?

Reply:
added + changed `asInternal` to use ComPtr